### PR TITLE
Fix: making token as optional field

### PR DESCRIPTION
**Pull request by NiharDoshi99**
### Describe your changes :

I worked on dagster making token as an optional field.
Fix : #9180 

#
### Type of change :

- [x] Bug fix
- [x] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

### CODE REVIEW

**Review by pmbrull — APPROVED** [review]

File: bootstrap/sql/com.mysql.cj.jdbc.Driver/v007__create_db_connection_info.sql [hunk 1]
```diff
@@ -36,3 +36,7 @@ CREATE TABLE IF NOT EXISTS alert_action_def (
     PRIMARY KEY (id),
     UNIQUE (name)
 );
+
+UPDATE pipeline_service_entity 
+SET json = JSON_REMOVE(json, '$.connection.config.configSource')
+WHERE serviceType = 'Dagster';
```
Comment (pmbrull) [inline bootstrap/sql/com.mysql.cj.jdbc.Driver/v007__create_db_connection_info.sql:42]: should we add the token for the services that we are removing it?

Comment (pmbrull) [reply bootstrap/sql/com.mysql.cj.jdbc.Driver/v007__create_db_connection_info.sql:42]: otherwise users will need to reenter it

Comment (NiharDoshi99) [reply bootstrap/sql/com.mysql.cj.jdbc.Driver/v007__create_db_connection_info.sql]: done. Thanks

File: ingestion/src/metadata/utils/connections.py [hunk 2]
```diff
@@ -1187,89 +1185,29 @@ def _(connection: DagsterClient) -> None:
...
 
-    url = config.host
-    cloud_dagster = DagsterGraphQLClient(
+    url = connection.host
+    dagster_connection = DagsterGraphQLClient(
         url,
         transport=RequestsHTTPTransport(
             url=f"{url}/graphql",
-            headers={"Dagster-Cloud-Api-Token": config.token.get_secret_value()},
+            headers={"Dagster-Cloud-Api-Token": connection.token.get_secret_value()}
```
Comment (ulixius9) [inline ingestion/src/metadata/utils/connections.py:1205]: this will break if the token is not passed

**Review by ulixius9 — APPROVED** [review]

**Pull request merged**